pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -196,50 +196,7 @@ func (t ResolvedPipelineTask) isFailure() bool {
 		c = t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
 		isDone = t.TaskRun.IsDone()
 	}
-	return isDone && c.IsFalse() && !t.hasRemainingRetries()
-}
-
-// hasRemainingRetries returns true only when the number of retries already attempted
-// is less than the number of retries allowed.
-func (t ResolvedPipelineTask) hasRemainingRetries() bool {
-	var retriesDone int
-	switch {
-	case t.IsCustomTask() && t.IsMatrixed():
-		if len(t.Runs) == 0 {
-			return true
-		}
-		// has remaining retries when any Run has a remaining retry
-		for _, run := range t.Runs {
-			retriesDone = len(run.Status.RetriesStatus)
-			if retriesDone < t.PipelineTask.Retries {
-				return true
-			}
-		}
-		return false
-	case t.IsCustomTask():
-		if t.Run == nil {
-			return true
-		}
-		retriesDone = len(t.Run.Status.RetriesStatus)
```

Comment:
This impacts custom tasks as well. Can the changes to custom tasks be split out into a separate PR?